app/scripts/controllers/preferences.js

```diff
@@ -247,7 +306,8 @@ class PreferencesController {
     } else {
       tokens.push(newEntry)
     }
-    this._updateAccountTokens(tokens)
+    assetImages[address] = image
```

Comment:
Not sure if it's really needed but maybe use toChecksumAddress or normalizeAddress to make sure it will always find the right key (assuming that assetImages keys are already checksummed)

Reply:
Yes, in line 295 the normalization happens, so for each token added it will have as key the correct normalized address.